src/expression.d

```diff
@@ -5255,6 +5257,66 @@ public:
                 continue;
             }
 
+            if (auto v = s.isVarDeclaration())
+            {
+                if ((!v.type || !v.type.deco) && v._scope)
+                    v.semantic(v._scope);
+
+                if (!v.type)
+                {
+                    error("forward reference of %s %s", v.kind(), v.toChars());
```

Comment:
This line error is already tested in [fail88.d](https://github.com/D-Programming-Language/dmd/blob/master/test/fail_compilation/fail88.d). And following error `recursive expansion` will be tested in #4731.